Userland/Libraries/LibWeb/HTML/TraversableNavigable.cpp

```diff
@@ -734,26 +734,32 @@ TraversableNavigable::HistoryStepResult TraversableNavigable::apply_the_history_
         // 9. Let entriesForNavigationAPI be the result of getting session history entries for the navigation API given navigable and targetStep.
         auto entries_for_navigation_api = get_session_history_entries_for_the_navigation_api(*navigable, target_step);
 
+        // NOTE: Steps 10 and 11 come after step 12.
+
         // 12. In both cases, let afterPotentialUnloads be the following steps:
         bool const update_only = changing_navigable_continuation->update_only;
         JS::GCPtr<SessionHistoryEntry> const target_entry = changing_navigable_continuation->target_entry;
+        JS::GCPtr<SessionHistoryEntry> const previous_entry = active_session_history_entry();
```

Comment:
Is this the correct way of doing this? I'm not sure if this could have changed by the time the `after_potential_unload` function runs.

Reply:
I am not sure - there is a bug somewhere relating to this function that I haven't quite worked our. I came across this spec bug relating to https://github.com/LadybirdBrowser/ladybird/pull/1808

Is there a particular reason why it is not done on step 1 in the afterPotentialReloadSteps and is put here?

Unfortunately something is broken with that with previousEntry last time I applied that spec change. It crashes the browser with https://github.com/LadybirdBrowser/ladybird/pull/1808 applied (which is where previousEntry is used)

Reply:
See: https://github.com/LadybirdBrowser/ladybird/pull/1808#discussion_r1800601621

Reply:
> Is there a particular reason why it is not done on step 1 in the afterPotentialReloadSteps and is put here?
I was assuming that capturing `this` would be Very Bad:tm:

Reply:
But yeah anyway, looks good to me!

Reply:
> > Is there a particular reason why it is not done on step 1 in the afterPotentialReloadSteps and is put here?
> 
> I was assuming that capturing `this` would be Very Bad ™️

Why would capturing `this` be very bad? Sounds like we're working with an incomplete mental model, can I help clarify something? :)

Reply:
I am indeed operating with almost 0 mental model. 😅 My understanding of what is or isn't safe to capture in a "run some time in the future" function is lacking.

Reply:
I think we would need a careful read of the spec to figure out if we should be using the "previous entry" from the check time, right now, or the use time, when we call after_potential_unload. In the case of step 10, we are calling it in a queued task on the NavigationAndTraversal task source. so it could happen many many cycles from now if there's a lot of other queue entries on that task source (such as other invocations of apply_the_history_step).

Reply:
I realised looking at it now that my code was wrong anyway, it wants `navigable`'s one, and we already capture `navigable`. So either my (new) change is safe, or the existing code was already unsafe.